modin/backends/pandas/query_compiler.py

```diff
@@ -343,7 +343,11 @@ def _append_list_of_managers(self, others, axis, **kwargs):
         join = kwargs.get("join", "outer")
         ignore_index = kwargs.get("ignore_index", False)
         new_self, to_append, joined_axis = self.copartition(
-            axis ^ 1, others, join, sort, force_repartition=True
+            axis ^ 1,
+            others,
+            join,
+            sort,
+            force_repartition=any(obj._is_transposed for obj in [self] + others),
```

Comment:
Don't we need to ensure that all the partitions have the same dimensions?

Reply:
That is ensured in `copartition`. The reason for the force repartitioning here is because the partitions could have different internal indices, which was fixed in this PR.